scoped-transitions.md

```diff
@@ -0,0 +1,110 @@
+# Scoped Transitions
+
+This doc is a summary of early design explorations to scope transitions to a DOM sub-tree. The following is a rough API sketch for how a developer would trigger the transition rooted on an element (other than the document's root element).
+
+```js
+element.createTransition({
+  updateDOM() {
+    /* … */
+  },
+});
+```
+
+The element becomes the **scoped-transition-root** for the transition. It's the element that will host the pseudo-element tree and page-transition-container sub-trees will be created for descendants that have a `page-transition-tag`. For example, the following page:
+
+```html
+<style>
+  .outer {
+    page-transition-tag: outer;
+    contain: layout;
+  }
+  .inner {
+    page-transition-tag: inner;
+    contain: layout;
+  }
+</style>
+<div class="outer">
+  <div class="inner"></div>
+</div>
+<script>
+  document.querySelector('.outer').createTransition(…);
+</script>
+```
+
+produces the following DOM tree under `outer`:
+
+```
+outer
+└─ ::page-transition
+   ├─ ::page-transition-container(outer)
+   │  └─ ::page-transition-image-wrapper(outer)
+   │     ├─ ::page-transition-outgoing-image(outer)
+   │     └─ ::page-transition-incoming-image(outer)
+   └─ ::page-transition-container(inner)
+      └─ ::page-transition-image-wrapper(inner)
+         ├─ ::page-transition-outgoing-image(inner)
+         └─ ::page-transition-incoming-image(inner)
+
+Note: We will probably need to change the naming, as 'page-transition' doesn't work.
+Note: We might want to add an ID for a transition to control which transition the `page-transition-tag` applies to.
+```
+
+The algorithm for executing the transition is as follows:
+
+1. At the next rendering opportunity after `createTransition`, the browser saves the painted output and geometry information for each tagged element under scoped-transition-root. The details are described in [7.3.4](https://drafts.csswg.org/css-shared-element-transitions-1/#perform-an-outgoing-capture-algorithm) except transforms are computed relative to the element. An async task is used to dispatch the `updateDOM` callback after this step.
```

Comment:
Is this the part that also describes render blocking? If we need an element level render blocking, I think we need to mention this explicitly, since it's one of the more complicated parts of the proposal

Reply:
The "Suppressing rendering" now pulls this discussion out separately too.